ui/src/createAzureClient.ts

```diff
@@ -3,52 +3,62 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { Environment } from '@azure/ms-rest-azure-env';
 import { BaseRequestPolicy, BasicAuthenticationCredentials, HttpOperationResponse, RequestPolicy, RequestPolicyFactory, RequestPolicyOptions, RequestPrepareOptions, RestError, ServiceClient, TokenCredentials, WebResource, WebResourceLike } from '@azure/ms-rest-js';
+import { v4 as uuid } from 'uuid';
 import * as vscode from "vscode";
 import * as types from '../index';
 import { appendExtensionUserAgent } from "./extensionUserAgent";
 import { GenericServiceClient } from './GenericServiceClient';
 import { localize } from './localize';
 import { maskValue } from './masking';
 import { parseError } from './parseError';
+import { AzExtTreeItem } from './tree/AzExtTreeItem';
 import { parseJson, removeBom } from './utils/parseJson';
 
-export function createAzureClient<T>(
-    clientInfo: { credentials: types.AzExtServiceClientCredentials; subscriptionId: string; environment: Environment; },
-    clientType: new (credentials: types.AzExtServiceClientCredentials, subscriptionId: string, options?: types.IMinimumServiceClientOptions) => T): T {
-    return new clientType(clientInfo.credentials, clientInfo.subscriptionId, {
+export type InternalAzExtClientContext = types.ISubscriptionActionContext | [types.IActionContext, types.ISubscriptionContext | AzExtTreeItem];
```

Comment:
I don't totally understand why we export the `InternalAzExtClientContext`, but not `type AzExtClientContext = ISubscriptionActionContext | [IActionContext, ISubscriptionContext | AzExtTreeItem];`  Why do we even need both since it doesn't seem like `AzExtClientContext` is exposed?

Reply:
The difference between `InternalAzExtClientContext` and `AzExtClientContext` is that the former references the actual `AzExtTreeItem` class in the `src/tree` folder and the latter references the class declaration from index.d.ts. I need to do this otherwise I get build errors related to `instanceof AzExTreeItem`

> it doesn't seem like AzExtClientContext is exposed?

That's...interesting, I assume you mean because I don't use `export` in index.d.ts? That was not intentional, and it actually still gets exported and can be referenced outside of this package. Maybe that's a bug in TypeScript? Lol anyways, I can add the `export` to make it explicit - there's a few others missing it as well.